src/misc/quiet-open.ts

```diff
@@ -1,32 +1,50 @@
-import { Door, Gate } from 'org.bukkit.block.data.type';
+import { Door, Gate, TrapDoor } from 'org.bukkit.block.data.type';
 import { Action } from 'org.bukkit.event.block';
 import { PlayerInteractEvent } from 'org.bukkit.event.player';
 import { Saw } from './saw';
 import { Material } from 'org.bukkit';
+import { EventPriority } from 'org.bukkit.event';
 
-registerEvent(PlayerInteractEvent, (event) => {
-  const block = event.clickedBlock;
-  if (event.action !== Action.RIGHT_CLICK_BLOCK) return;
-  if (!block) return;
-  if (event.player.isSneaking()) return;
+registerEvent(
+  PlayerInteractEvent,
+  (event) => {
+    const block = event.clickedBlock;
+    if (event.action !== Action.RIGHT_CLICK_BLOCK) return;
+    if (!block) return;
+    if (event.player.isSneaking()) return;
+    if (event.isCancelled()) return;
+    const blockData = block.blockData;
+    let sound = '';
 
-  // Wooden doors
-  if (block.blockData instanceof Door) {
-    if (block.type === Material.IRON_DOOR) return;
-    // Custom sound names with vanilla sound files (sounds.json)
-    const sound = (block.blockData as Door).isOpen()
-      ? 'non-silent.wooden_door.close'
-      : 'non-silent.wooden_door.open';
-    block.world.playSound(block.location, sound, 1, 1);
-  }
+    // Wooden doors
+    if (blockData instanceof Door) {
+      if (block.type === Material.IRON_DOOR) return;
+      sound = blockData.isOpen()
+        ? 'non-silent.wooden_door.close'
+        : 'non-silent.wooden_door.open';
+    }
 
-  // Fence gates
-  else if (block.blockData instanceof Gate) {
-    if (Saw.check(block)) return;
-    // Custom sound names with vanilla sound files (sounds.json)
-    const sound = (block.blockData as Gate).isOpen()
-      ? 'non-silent.fence_gate.close'
-      : 'non-silent.fence_gate.open';
-    block.world.playSound(block.location, sound, 1, 1);
-  }
-});
+    // Fence gates
+    else if (blockData instanceof Gate) {
+      if (Saw.check(block)) return;
+      sound = blockData.isOpen()
+        ? 'non-silent.fence_gate.close'
+        : 'non-silent.fence_gate.open';
+    }
+
+    // Fence gates
+    else if (blockData instanceof TrapDoor) {
+      sound = blockData.isOpen()
+        ? 'non-silent.wooden_trapdoor.close'
+        : 'non-silent.wooden_trapdoor.open';
+    }
+
+    if (sound) block.world.playSound(block.location, sound, 1, 1);
```

Comment:
```suggestion
    if (sound !== '') block.world.playSound(block.location, sound, 1, 1);
```
Even if empty string is falsy, I think this should be a not equals.

Reply:
Agreed. If the string could also be e.g. undefined or null, then this would make more sense.